src/libsystemd/sd-bus/bus-socket.c

```diff
@@ -152,6 +154,14 @@ static int bus_socket_write_auth(sd_bus *b) {
                 return ERRNO_IS_TRANSIENT(errno) ? 0 : -errno;
 
         iovec_advance(b->auth_iovec, &b->auth_index, (size_t) k);
+
+        if (b->is_server && !bus_socket_auth_needs_write(b)) {
+                // When everything has been written, we might process "BEGIN" now.
+                int r = bus_socket_auth_verify_server(b);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
hmm, I think instead of making this specific to the server mode, we should do this for either mode. i.e. drop the `is_server` check, and call `bus_socket_auth_verify()`.

Also `int r` is special to us, as the primary variable to store function return codes in, as such it is always defined at the outermost scope of the funciton, hence please move `int r;` to the to of the function.

Hmm, and then please use `/* */` style comments, we use `//` comments for FIXME style comments only.

See https://systemd.io/CODING_STYLE/#formatting for details.

Reply:
Ok! I have made it even simpler, by calling `bus_socket_auth_verify` unconditionally after a successful write. That should be ok as well, right?